lib/Attribute.js

```diff
@@ -553,8 +553,12 @@ Attribute.prototype.parseDynamo = function(json) {
     val = dedynamofy('N', this.isSet, json, datify, this);
     break;
     case 'object':
-    val = dedynamofy('S', this.isSet, json, JSON.parse, this);
-    break;
+      if(json.M) {
+        val = dedynamofy('M', this.isSet, json, mapify, this);
+      } else {
+        val = dedynamofy('S', this.isSet, json, JSON.parse);
```

Comment:
Is there a reason why the line that was removed included `this` as the last parameter but not this new version?

Reply:
100% of tests pass with this change. I can guarantee some code fails without it since that's how I found it.  Let me see what I can do to prove it.

Reply:
@brennhill Yeah I'm just looking for new tests to be added. So in case this bug slips in again our tests will catch it before hand. If that makes sense.

Reply:
I went and looked at adding a test and it's going to take a lot of work to reproduce because dynamo stores objects as JSON strings intead of MAP.  Which means i'd have to fix how dynamo stores data, which would be a beaking change.  

As stated in my decription, this is a fix for circumstances where a process *other* than dynamoose is doing the saves to dynamo and is correctly saving Objects as Maps insted of JSON strings.